[CI] JwkSetLoaderTests testConcurrentReloadWillBeQueuedAndShareTheResults failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/5g254tjz37xwo/tests/:x-pack:plugin:security:test/org.elasticsearch.xpack.security.authc.jwt.JwkSetLoaderTests/testConcurrentReloadWillBeQueuedAndShareTheResults

**Reproduction line:**
```
./gradlew ':x-pack:plugin:security:test' --tests "org.elasticsearch.xpack.security.authc.jwt.JwkSetLoaderTests.testConcurrentReloadWillBeQueuedAndShareTheResults" -Dtests.seed=1EF0077A5061E504 -Dtests.locale=id -Dtests.timezone=Africa/Dar_es_Salaam -Druntime.java=17
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.security.authc.jwt.JwkSetLoaderTests&tests.test=testConcurrentReloadWillBeQueuedAndShareTheResults

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: sameInstance(<Tuple[v1=false, v2=JwksAlgs[jwks=[], algs=[]]]>)
     but: was <Tuple[v1=false, v2=JwksAlgs[jwks=[], algs=[]]]>

  at __randomizedtesting.SeedInfo.seed([1EF0077A5061E504:57606554A5F0FEA9]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.xpack.security.authc.jwt.JwkSetLoaderTests.lambda$testConcurrentReloadWillBeQueuedAndShareTheResults$4(JwkSetLoaderTests.java:71)
  at java.lang.Iterable.forEach(Iterable.java:75)
  at org.elasticsearch.xpack.security.authc.jwt.JwkSetLoaderTests.testConcurrentReloadWillBeQueuedAndShareTheResults(JwkSetLoaderTests.java:71)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwkSetLoaderTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwkSetLoaderTests.java
@@ -13,6 +13,7 @@ import org.elasticsearch.env.Environment;
 import org.elasticsearch.test.ESTestCase;
 import org.elasticsearch.xpack.core.security.authc.RealmConfig;
 import org.elasticsearch.xpack.core.security.authc.jwt.JwtRealmSettings;
+import org.mockito.Mockito;
 
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
@@ -28,14 +29,13 @@ import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.sameInstance;
 import static org.mockito.Mockito.doAnswer;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.spy;
-import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 public class JwkSetLoaderTests extends ESTestCase {
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/92745")
     public void testConcurrentReloadWillBeQueuedAndShareTheResults() throws IOException, InterruptedException {
         final Path tempDir = createTempDir();
         final Path path = tempDir.resolve("jwkset.json");
@@ -53,23 +53,37 @@ public class JwkSetLoaderTests extends ESTestCase {
             .mapToObj(i -> new PlainActionFuture<Tuple<Boolean, JwkSetLoader.JwksAlgs>>())
             .toList();
 
-        final var threadsCountDown = new CountDownLatch(nThreads);
+        // Start the first thread for reloading
+        // Ensure it is inside the actual loading method and make it wait there to simulate slow processing
+        final var loadingLatch = new CountDownLatch(1);
         final var readyLatch = new CountDownLatch(1);
         doAnswer(invocation -> {
-            // Mark the thread to be ready when it enters the concurrency controlling area
-            threadsCountDown.countDown();
-            // Wait till the ready flag to start the racing
+            loadingLatch.countDown();
             assertThat(readyLatch.await(10, TimeUnit.SECONDS), is(true));
-            return invocation.callRealMethod();
-        }).when(jwkSetLoader).getFuture();
+            invocation.callRealMethod();
+            return null;
+        }).when(jwkSetLoader).loadInternal(anyActionListener());
 
-        // Start the threads and toggle flags to start racing
-        IntStream.range(0, nThreads).forEach(i -> new Thread(() -> jwkSetLoader.reload(futures.get(i))).start());
+        new Thread(() -> jwkSetLoader.reload(futures.get(0))).start();
+        assertThat(loadingLatch.await(10, TimeUnit.SECONDS), is(true));
+
+        // Start rest of the threads for racing and ensure they are all through the concurrency controlling area
+        Mockito.reset(jwkSetLoader);
+        final var threadsCountDown = new CountDownLatch(nThreads - 1);
+        doAnswer(invocation -> {
+            final Object result = invocation.callRealMethod();
+            threadsCountDown.countDown();
+            return result;
+        }).when(jwkSetLoader).getFuture();
+        IntStream.range(1, nThreads).forEach(i -> new Thread(() -> jwkSetLoader.reload(futures.get(i))).start());
         assertThat(threadsCountDown.await(10, TimeUnit.SECONDS), is(true));
+
+        // Notify the first thread to finish the loading work
         readyLatch.countDown();
 
-        // All concurrent reloading calls will get the same result and the actual reloading work will only happen once
-        futures.forEach(future -> assertThat(future.actionGet(), sameInstance(futures.get(0).actionGet())));
-        verify(jwkSetLoader, times(1)).loadInternal(anyActionListener());
+        // All concurrent reloading calls will get the same result from the first thread and skip the actual loading work
+        final Tuple<Boolean, JwkSetLoader.JwksAlgs> tuple = futures.get(0).actionGet();
+        futures.subList(1, nThreads).forEach(future -> assertThat(future.actionGet(), sameInstance(tuple)));
+        verify(jwkSetLoader, never()).loadInternal(anyActionListener());
     }
 }

